model/src/main/kotlin/io/spine/validation/ErrorPlaceholders.kt

```diff
@@ -103,8 +103,8 @@ internal fun String.checkPlaceholders(
 )
 
 /**
- * Checks if this [String] contains placeholders that are not present
- * in the given set of the [supported], and reports a compilation error if so.
+ * Checks if this [String] contains placeholders that are not present in the given
+ * set of the [supported] placeholders, and reports a compilation error if so.
  */
 private fun String.checkPlaceholders(
```

Comment:
If memory serves, we have the `TemplateString` type. Why isn't it its responsibility to check placeholders?

I'm not too worried about having extensions to the `String` type as long as they are `private` or `internal`. But it looks a bit suspicious.

Reply:
`TemplateString` contains the message template along with its placeholders, and values for them. And it doesn't use any enums, placeholders are just strings.

There are extensions that check it and throw in case of unexpected placeholders.

But for policies, we need to check it without values (values appear later, in runtime) and report a compilation error. And it is easier to do with `ErrorPlaceholder` enum which we have for `:model` and `:java`.